Code/Source/Entity/MeshEntityManager.cpp

```diff
@@ -69,6 +70,11 @@ namespace RGL
                     entity.SetIntensityTexture(texture);
                 }
 
+                if (m_packedRglEntityId.has_value())
+                {
+                    entity.SetId(m_packedRglEntityId.value());
```

Comment:
I am curious about one thing here.

First, m_packedRglEntityId comes from EntityManager and is set inside OnSegmentationClassesReady that is called from ClassSegmentationConnectionPolicy::Connect.

Then, this id is set here, in OnModelReady. This is probably correct, but do we know that OnModelReady will always be called after OnSegmentationClassesReady? E.g. based on some engine execution order? Or based on some other trait?

The same question referes to ActorEntityManager too.

Reply:
I've modified Actor and Mesh entity managers for the exact reason that the entities may be created after the id is initialized. Also in case where the entities are already present:
![image](https://github.com/user-attachments/assets/689a893d-0d36-4c78-87be-63dd64ce2f47)
I'm pretty sure but not certain that the `OnSegmentationClassesReady` will be called before any entities are constructed. This would be due to the fact that this is called after the level entity is activated. Regardless of this I've made it so that the activation order doesn't matter.